frontend/src/metabase/components/DebouncedFrame.jsx

```diff
@@ -49,11 +55,11 @@ export default class DebouncedFrame extends React.Component {
       this.state.height !== nextProps.height
     ) {
       if (this.state.width == null || this.state.height == null) {
-        this.setSizeDebounced(nextProps.width, nextProps.height);
+        this.updateSizeDebounced();
```

Comment:
The main change is that instead of specifying what size the state should be set to in the future (because the function is debounced), use the present information at the time the frame would be resized.

This fixes the issue @flamber described in https://github.com/metabase/metabase/pull/20954#issuecomment-1069541046

In that case, after adding a log inside `UNSAFE_componentWillReceiveProps` lifecycle

![image](https://user-images.githubusercontent.com/1937582/158761247-6f6117a7-b93c-42c4-bb06-e0c737058cbf.png)

You could clearly see that somehow (not related to this PR) the height changed from a smaller value (accounting the footer height) to a larger value (without accounting for the footer height) and back to a smaller value eventually.

The problem with the previous code is that the state is decided whether or not it needs to be changed in advance. On the 3rd log when props differs from the state (398px vs 349px), the state is scheduled to be changed. But when the 4th log came, height was still the same and has the same value as the prpos since state changed is still being scheduled (from debounced), so there will be no state change schedule at this point. Then after debounced periord end the height will be set to `398px` which make it out of sync with the state.

The fix is to determine the width and height we want to set to the state at the time we're about to set the state. This would be more reliable as determining the state in advance is flaky as I've described.